vendor/k8s.io/kubernetes/test/e2e/scheduling/preemption.go

```diff
@@ -90,45 +90,67 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
 		var podRes v1.ResourceList
 		// Create one pod per node that uses a lot of the node's resources.
 		By("Create pods that use 60% of node resources.")
-		pods := make([]*v1.Pod, len(nodeList.Items))
+		pods := make([]*v1.Pod, 0)
 		allPods, err := cs.CoreV1().Pods(metav1.NamespaceAll).List(metav1.ListOptions{})
+		framework.ExpectNoError(err)
 		for i, node := range nodeList.Items {
 			currentCpuUsage, currentMemUsage := getCurrentPodUsageOnTheNode(node.Name, allPods.Items, podRequestedResource)
 			framework.Logf("Current cpu and memory usage %v, %v", currentCpuUsage, currentMemUsage)
-			currentNode, err := cs.CoreV1().Nodes().Get(node.Name, metav1.GetOptions{})
-			framework.ExpectNoError(err)
-			cpuAllocatable, found := currentNode.Status.Allocatable["cpu"]
+			cpuAllocatable, found := node.Status.Allocatable["cpu"]
 			Expect(found).To(Equal(true))
 			milliCPU := cpuAllocatable.MilliValue()
-			milliCPU = milliCPU * 40 / 100
-			memAllocatable, found := currentNode.Status.Allocatable["memory"]
+			// Just to be tolerant use 0.6 of resources available on the node
+			milliCPU = int64(float64(milliCPU-currentCpuUsage) * float64(0.6))
+			memAllocatable, found := node.Status.Allocatable["memory"]
 			Expect(found).To(Equal(true))
 			memory := memAllocatable.Value()
-			memory = memory * 60 / 100
+			// Just to be tolerant use 0.6 of resources available on the node
+			memory = int64(float64(memory-currentMemUsage) * float64(0.6))
 			podRes = v1.ResourceList{}
+			// If a node is already heavily utilized let not's create a pod there.
+			if milliCPU <= 0 {
+				framework.Logf("Node is heavily utilized, let's not create a pod here")
+				continue
+			}
 			podRes[v1.ResourceCPU] = *resource.NewMilliQuantity(int64(milliCPU), resource.DecimalSI)
-			podRes[v1.ResourceMemory] = *resource.NewQuantity(int64(memory), resource.BinarySI)
```

Comment:
The test is currently failing because of this change and I will revert it. Without this we were getting the following error:

`skip [k8s.io/kubernetes/test/e2e/scheduling/preemption.go:234]: We need atleast two pods to be created butall nodes are already heavily utilized, so preemption tests cannot be run`

Perhaps we need to increase the node size now since we're skipping those tests?